src/main/java/org/jnode/fs/ntfs/datarun/CompressedDataRun.java

```diff
@@ -220,43 +248,96 @@ private static int uncompressBlock(final OffsetByteArray compressed,
             return 0;
         }
 
+        // Bit 15 indicates whether the chunk contains compressed data. If this bit is zero, the chunk header is followed by uncompressed literal data.
+        // If this bit is set, the next byte of the chunk is the beginning of a Flag_group nonterminal that describes some compressed data.
         if ((rawLen & 0x8000) == 0) {
             // Uncompressed chunks store length as 0xFFF always.
             if ((len + 1) != BLOCK_SIZE) {
                 log.debug("ntfs_uncompblock: len: " + len + " instead of 0xfff");
             }
 
-            // Copies the entire compression block as-is, need to skip the compression flag,
-            // no idea why they even stored it given that it isn't used.
-            // Darwin's version I was referring to doesn't skip this, which seems be a bug.
-            uncompressed.copyFrom(compressed, cpos, 0, len + 1);
-            uncompressed.zero(len + 1, BLOCK_SIZE - 1 - len);
```

Comment:
Is this necessary? Needs more tests to confirm whether it still needs to be filled with zero or not.
hmmm, maybe still yes.. then I need to rollback it here.